### PR TITLE
Users should be able to login/logout and use the application.

### PR DESCRIPTION
**Pull request by azaria-dedmon**
**What does this PR do?**

- This PR provides functionality for users to login/logout and use the application.

**Description of Task to be completed.**

- Add authentication method to `User` table.
- Add `login` and `logout` endpoints.
- Include user homepage endpoint to test login/logout functionality.
- Include relevant unit tests.
- Includes some signup functionality in order to be able to create test user for login/logout.

**How should this be manually tested?**

- Run `Python3 run.py`
- Go to `localhost:5000/login`

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,6 +1,94 @@
...
+                vax_date=form.vax_date.data,
+                covid_status=form.covid_status.data)
+        db.session.commit()
+        if user:
+            do_login(user)
+            return redirect('/user')
+        flash("Username already taken", 'danger')
+        return render_template('users/register.html', form=form)
+
+    return render_template('users/register.html', form=form)
```
Comment (alexkayabula) [inline app.py]: @azaria-dedmon how is this related to login/logout.

File: forms.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    """Form for registering users"""
+    firstname = StringField('First Name', validators=[DataRequired()])
+    lastname = StringField('Last Name', validators=[DataRequired()])
+    username = StringField('Username', validators=[DataRequired()])
+    email = StringField('Email', validators=[DataRequired()])
+    password = PasswordField('Password', validators=[DataRequired()])
+    image = StringField('Image Url')
+    state = SelectField('State', choices=testing_states, validators=[DataRequired()])
+    vax_date = StringField('Vaccination Date')
+    covid_status = StringField('Covid Status')
```
Comment (alexkayabula) [inline forms.py]: @azaria-dedmon how is this related to login/logout ?

File: models.py [hunk 3]
```diff
@@ -56,8 +55,40 @@ class User (db.Model):
...
+            username=username,
+            email=email,
+            password=hashed_pwd,
+            image=image,
+            state=state,
+            vax_date=vax_date,
+            covid_status=covid_status)
+
+        db.session.add(user)
+        return user
```
Comment (alexkayabula) [inline models.py]: @azaria-dedmon how is this related to login/logout ?

File: tests/test_user_model.py [hunk 4]
```diff
@@ -58,3 +61,61 @@ def test_user_model(self):
...
+        db.session.commit()
+
+        user = User.authenticate('test123', 'password')
+        db.session.commit()
+
+        self.assertEqual(user.username, 'test123')
+
+
+    def test_invalid_username_authentication(self):
+        self.assertFalse(User.authenticate("badusername", "password"))
```
Comment (alexkayabula) [inline tests/test_user_model.py]: How is this related to login/logout ?

File: app.py [hunk 5]
```diff
@@ -1,6 +1,94 @@
...
+from flask import Flask, session, g, render_template, redirect, flash
+import requests
+from forms import RegisterUser, LoginUser
+from models import connect_db, User, db
+
+
+CURR_USER_KEY = "curr_user"
 
 app = Flask(__name__)
+app.config['SECRET_KEY'] = "application20212021"
```
Comment (alexkayabula) [inline app.py]: @azaria-dedmon use environment variables for secrets.

File: models.py (hunk 3 above)
Comment (azaria-dedmon) [reply models.py]: I have to include the signup functionality in order to properly test for the login/logout feature. To test if a user is able to login/logout, I have to first register an account.

**Comment by alexkayabula** [comment]
@azaria-dedmon please resolve the merge conflicts.

**Comment by alexkayabula** [comment]
@azaria-dedmon this PR is going to revert some of the already existing functionality(Some of the register user functionality). This could have have been caused by some fault during rebase/ while resolving merge conflicts.

**Comment by alexkayabula** [comment]
@azaria-dedmon, delete this pull request. 
1. Create a new pull request from master. 
2. Add the implementation.
3. Make ONLY one commit message.
